### PR TITLE
Add missing tzdata in alpine 3.13 and 3.14 amd64

### PR DESCRIPTION
**Pull request by am11**
@ViktorHofer, this one got missed in the refactoring, which is why the ICU test was failing to find `Europe/London` timezone (amd64-only issue).

### CODE REVIEW

**Comment by am11** [comment]
In second commit, cleaned up some sub-dependencies which we don't (directly) need and unified the files so they "look" a like (would make it easier in future to deduplicate this code when we switch to templates like https://github.com/dotnet/dotnet-docker/blob/44483abdb1bd260048d8efc6eb56bc56b66e8ea2/eng/dockerfile-templates/runtime/6.0/Dockerfile.linux).

File: src/alpine/3.13/WithNode/amd64/Dockerfile [hunk 1]
```diff
@@ -1,34 +1,9 @@
...
-            && curl -fsSLO --compressed "https://yarnpkg.com/downloads/$YARN_VERSION/yarn-v$YARN_VERSION.tar.gz" \
-            && curl -fsSLO --compressed "https://yarnpkg.com/downloads/$YARN_VERSION/yarn-v$YARN_VERSION.tar.gz.asc" \
-            && gpg --batch --verify yarn-v$YARN_VERSION.tar.gz.asc yarn-v$YARN_VERSION.tar.gz \
-            && mkdir -p /opt \
-            && tar -xzf yarn-v$YARN_VERSION.tar.gz -C /opt/ \
-            && ln -s /opt/yarn-v$YARN_VERSION/bin/yarn /usr/local/bin/yarn \
-            && ln -s /opt/yarn-v$YARN_VERSION/bin/yarnpkg /usr/local/bin/yarnpkg \
-            && rm yarn-v$YARN_VERSION.tar.gz.asc yarn-v$YARN_VERSION.tar.gz \
-            && apk del .build-deps-yarn
+RUN apk update && apk add --no-cache nodejs yarn
```
Comment (mthalman) [inline src/alpine/3.13/WithNode/amd64/Dockerfile:3]: Recommended to list out each package name on its own line.

**Review by mthalman — APPROVED** [review]

File: src/alpine/3.13/WithNode/amd64/Dockerfile (hunk 1 above)
Comment (am11) [reply src/alpine/3.13/WithNode/amd64/Dockerfile]: Updated.

**Comment by am11** [comment]
@ViktorHofer, if there is no other feedback could this be merged? I'll update runtime PR to fix the culture failing tests.

**Pull request merged**